### ISSUE TITLE
There should be a way to disable student emails for an individual course

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#3259]
Occasionally an instructor wants to stop any reminder emails for trainings to be sent to students.

We could perhaps implement it as a course flag, which would get checked before the training reminder emails go out.

**Comment by marklocklear** [comment]
A couple of notes on this one:

- add `disable_training_emails` boolean flag to course.rb model
- add logic to `overdue_training_alert.rb` with something like:
```
def send_email
    return if emails_disabled?
    return if course.disable_training_emails?
    return if opted_out?
    OverdueTrainingAlertMailer.send_email(self)
  end
```

- Add drop-down selection to Details page of a course allowing the instructor to disable training emails for students

@ragesoss other alerts/mailers I should consider for this feature?

**Comment by ragesoss** [comment]
I think think there are other alerts that should be wrapped in as well (so, the flag and selector shouldn't be specific to training emails).

When an instructor wants to turn off emails for their students, they typically want to try to ensure that NO email from the dashboard reaches their students at all. The relevant one that I spot from `/rails/mailers` is: `High Quality Article Assignment Mailer`. I imagine there would be others in the future; we don't currently send very many emails to students, but I think we will add more over time.

I'm not sure what the right way is to capture in code the idea of "this set of emails should not go out to students, if their course has student emails disabled".

**Comment by marklocklear** [comment]
Ah, I see. Something more abstract would be adding a boolean `disabled` field to the alert model. Each course would have an instance of each of its alerts (OverdueTrainingAlert, ProductiveCourseAlert...etc) then you could enable/disable each of those alerts if you like. Default would be true, then in each of the alert models (ie class OverdueTrainingAlert < Alert) we could call `return if self.disabled?` in the `send_email` method.

Just FYI I do see:

``` 
  def emails_disabled?
    ENV["#{self.class}_emails_disabled"] == 'true'
  end
```

...in alert.rb, but this is system-level email disabling for all courses?

**Comment by ragesoss** [comment]
Correct, that is code you found is for system-level disabling of emails for an alert type.

The `flags` hash for an individual course is probably the right place for this to live. I don't think we want to disable alert-types one-by-one at the course level, just one toggle for whether students should receive any course-related emails or not.

**Comment by marklocklear** [comment]
OK, first stab at this here: https://github.com/marklocklear/WikiEduDashboard/tree/disable_email_for_course. I only added a single check for the flag in OverdueTrainingAlert, but can add other places depending on the need.

**Comment by ragesoss** [comment]
Cool. Open a PR?
